### PR TITLE
🎨  Update Table to use forwardRef

### LINKED ISSUE #810: Conform on the use of forwardRef and {...rest}
Some component (at least the `Snackbar`, `Banner` and `Table` components) are missing forwardRef and spreading of rest props. This should be aligned across all components.

Also, some component use {...props} at the last parameter, other as the first as some logic depends on another order.

### LINKED ISSUE #1007: Update Table to use forwardRef
Update Table and its sub components to use, React's `forwardRef` so its on par with our other components

### PR DESCRIPTION
**Pull request by pomfrida**
Resolves #1007

### CODE REVIEW

File: libraries/core-react/src/components/Table/Head.tsx [hunk 1]
```diff
@@ -10,12 +10,12 @@ const StyledTableHead = styled.thead`
   background: ${token.background};
 `
 
-export const Head: FunctionComponent = ({ children, ...props }) => {
+type HeadProps = { children: ReactNode }
```
Comment (mimarz) [inline libraries/core-react/src/components/Table/Head.tsx:13]: Should this not be `React.HTMLAttributes<HTMLTableHeaderCellElement>` type?

File: libraries/core-react/src/components/Table/Head.tsx [hunk 2]
```diff
@@ -10,12 +10,12 @@ const StyledTableHead = styled.thead`
...
+
+export const Head = ({ children, ...props }: HeadProps): JSX.Element => {
```
Comment (mimarz) [inline libraries/core-react/src/components/Table/Head.tsx:15]: Missing use of `forwardRef` here

**Review by mimarz — CHANGES_REQUESTED** [review]
Seems to be some missing changes in `Cell` and `Body` for `forwardRef`?

**Comment by pomfrida** [comment]
@mimarz I had some problems with the `ref` and the `TableHeaderCellElement` type in `Head`. It suggested to use `HTMLTableSectionElement` instead. Last commit with `Body` and `Head` fix 👍

**Review by mimarz — APPROVED** [review]
LGTM 👍

**Pull request merged**